### ISSUE TITLE
Use `Iterator::sum` instead of `fold`

### ISSUE THREAD

**Issue by tomassedovic** [tomassedovic/roguelike-tutorial#11]
Rust 1.11.0 has finally stabilised the `sum` iterator method:

https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.sum

That means we can change code like this:

```
let bonus = self.get_all_equipped(game).iter().fold(0, |sum, e| sum + e.power_bonus);
```

to something like:

```
let bonus = self.get_all_equipped(game).iter().map(|e| e.power_bonus).sum()
```

Fold is awesome, but this should be easier to explain & understand.

https://github.com/tomassedovic/roguelike-tutorial/blob/11e50b1939b101a0adffa4e4c6fef957a20f593f/src/bin/part-13-adventure-gear.rs#L273-L285

**Comment by iszla** [comment]
I added this in #52 as well

**Comment by tomassedovic** [comment]
Fixed in #52
